### PR TITLE
Set app icon ut coverage.

### PR DESCRIPTION
**Pull request by BSolonenko**
This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
This PR provides unit test for set appIcon when app register.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by BSolonenko** [comment]
@AKalinich-Luxoft, @AByzhynar and @Ypostolov Please review.

**Comment by BSolonenko** [comment]
Rebuild required

File: src/components/application_manager/test/application_manager_impl_test.cc [hunk 1]
```diff
@@ -941,6 +946,49 @@ TEST_F(ApplicationManagerImplTest, UnregisterAnotherAppDuringAudioPassThru) {
   }
 }
 
+TEST_F(ApplicationManagerImplTest,
+       RegisterApplication_TrueIconPath_PathToTheIconWasSet) {
+  file_system::CreateDirectory(kDirectoryName);
```
Comment (AByzhynar) [inline src/components/application_manager/test/application_manager_impl_test.cc]: @BSolonenko All these functions return bool. Check returned values before further processing

File: src/components/application_manager/test/application_manager_impl_test.cc [hunk 2]
```diff
@@ -941,6 +946,49 @@ TEST_F(ApplicationManagerImplTest, UnregisterAnotherAppDuringAudioPassThru) {
...
+
+  request_for_registration[strings::params][strings::protocol_version] =
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2;
+
+  smart_objects::SmartObjectSPtr request_for_registrationSPtr =
+      MakeShared<smart_objects::SmartObject>(request_for_registration);
+
+  ApplicationSharedPtr application =
+      app_manager_impl_->RegisterApplication(request_for_registrationSPtr);
+  EXPECT_STREQ(kAppName.c_str(), application->name().c_str());
```
Comment (AByzhynar) [inline src/components/application_manager/test/application_manager_impl_test.cc]: @BSolonenko Compare without redundant conversation

File: src/components/application_manager/test/application_manager_impl_test.cc [hunk 3]
```diff
@@ -941,6 +946,49 @@ TEST_F(ApplicationManagerImplTest, UnregisterAnotherAppDuringAudioPassThru) {
   }
 }
 
+TEST_F(ApplicationManagerImplTest,
+       RegisterApplication_TrueIconPath_PathToTheIconWasSet) {
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply src/components/application_manager/test/application_manager_impl_test.cc]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2121/commits/d3669007e038ee0921f3d6fa4b3786b723fed263

File: src/components/application_manager/test/application_manager_impl_test.cc (hunk 1 above)
Comment (BSolonenko) [reply src/components/application_manager/test/application_manager_impl_test.cc]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2121/commits/d3669007e038ee0921f3d6fa4b3786b723fed263

File: src/components/application_manager/test/application_manager_impl_test.cc [hunk 4]
```diff
@@ -941,6 +946,49 @@ TEST_F(ApplicationManagerImplTest, UnregisterAnotherAppDuringAudioPassThru) {
...
+  request_for_registration[strings::msg_params][strings::app_name] = kAppName;
+  request_for_registration[strings::msg_params][strings::sync_msg_version]
+                          [strings::minor_version] = APIVersion::kAPIV2;
+  request_for_registration[strings::msg_params][strings::sync_msg_version]
+                          [strings::major_version] = APIVersion::kAPIV3;
+
+  request_for_registration[strings::params][strings::protocol_version] =
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2;
+
+  smart_objects::SmartObjectSPtr request_for_registrationSPtr =
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply src/components/application_manager/test/application_manager_impl_test.cc]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2121/commits/d3669007e038ee0921f3d6fa4b3786b723fed263

File: src/components/application_manager/test/application_manager_impl_test.cc (hunk 2 above)
Comment (BSolonenko) [reply src/components/application_manager/test/application_manager_impl_test.cc]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2121/commits/d3669007e038ee0921f3d6fa4b3786b723fed263

File: src/components/application_manager/test/application_manager_impl_test.cc [hunk 5]
```diff
@@ -941,6 +946,49 @@ TEST_F(ApplicationManagerImplTest, UnregisterAnotherAppDuringAudioPassThru) {
...
+  EXPECT_STREQ(kAppName.c_str(), application->name().c_str());
+  EXPECT_STREQ(full_icon_path.c_str(), application->app_icon_path().c_str());
+  EXPECT_EQ(protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2,
+            application->protocol_version());
+  EXPECT_EQ(APIVersion::kAPIV2,
+            application->version().min_supported_api_version);
+  EXPECT_EQ(APIVersion::kAPIV3,
+            application->version().max_supported_api_version);
+
+  file_system::RemoveDirectory(kDirectoryName, true);
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply src/components/application_manager/test/application_manager_impl_test.cc]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2121/commits/d3669007e038ee0921f3d6fa4b3786b723fed263

**Pull request merged**